src/openms/source/FORMAT/CachedMzML.cpp

```diff
@@ -357,13 +474,42 @@ namespace OpenMS
 
     Datavector rt_data;
     Datavector int_data;
+    rt_data.reserve(chromatogram.size());
+    int_data.reserve(chromatogram.size());
     for (Size j = 0; j < chromatogram.size(); j++)
     {
       rt_data.push_back(chromatogram[j].getRT());
       int_data.push_back(chromatogram[j].getIntensity());
     }
     ofs.write((char*)&rt_data.front(), rt_data.size() * sizeof(rt_data.front()));
     ofs.write((char*)&int_data.front(), int_data.size() * sizeof(int_data.front()));
+
+    for (const auto& fda : chromatogram.getFloatDataArrays() )
+    {
+      Size len = fda.size();
+      ofs.write((char*)&len, sizeof(len));
+      Size len_name = fda.getName().size();
+      ofs.write((char*)&len_name, sizeof(len_name));
+      ofs.write((char*)&fda.getName().front(), len_name * sizeof(fda.getName().front()));
+      // now go to the actual data
+      Datavector tmp;
+      tmp.reserve(fda.size());
+      for (const auto& val : fda) {tmp.push_back(val);}
+      ofs.write((char*)&tmp.front(), tmp.size() * sizeof(tmp.front()));
+    }
+    for (const auto& ida : chromatogram.getIntegerDataArrays() )
+    {
+      Size len = ida.size();
+      ofs.write((char*)&len, sizeof(len));
+      Size len_name = ida.getName().size();
+      ofs.write((char*)&len_name, sizeof(len_name));
+      ofs.write((char*)&ida.getName().front(), len_name * sizeof(ida.getName().front()));
+      // now go to the actual data
+      Datavector tmp;
+      tmp.reserve(ida.size());
```

Comment:
optimization:
instead of allocating tmp in every loop, you could declare it outside, while still calling reserve() + clear() inside before pushing. Should be faster, since it saves the (de)allocation.